src/mcp_atlassian/server.py

```diff
@@ -89,23 +166,183 @@ async def server_lifespan(server: Server) -> AsyncIterator[AppContext]:
             jira_url = jira.config.url
             logger.info(f"Jira URL: {jira_url}")
 
+        logger.debug("Server lifespan initialized")
+
         # Provide context to the application
         yield AppContext(confluence=confluence, jira=jira)
     finally:
         # Cleanup resources if needed
+        logger.debug("Server lifespan ending")
         pass
 
 
 # Create server instance
 app = Server("mcp-atlassian", lifespan=server_lifespan)
 
 
+# Session tracking for long-lived connections
+_session_last_activity: dict[str, float] = {}
+_detected_client_types: dict[str, str] = {}  # Maps session_id to client_type
+
+
+def get_client_type() -> str:
+    """Get the detected client type from the app context or detect it."""
+    # Check for forced Cline mode via environment variable
+    if os.getenv("MCP_ATLASSIAN_FORCE_CLINE", "0") == "1":
```

Comment:
Environment variables like `MCP_ATLASSIAN_FORCE_CLINE` create hidden behavior that's difficult to understand and maintain.